app/src/main/java/com/example/filemanager/AllFilesFragment.kt

```diff
@@ -56,7 +60,63 @@ class AllFilesFragment : Fragment(), ItemClickListener {
         })
     }
 
+    override fun onStart() {
+        super.onStart()
+        fetchFiles()
+    }
+
     override fun onItemClickListener() {
         Toast.makeText(context, "Item clicked", Toast.LENGTH_SHORT).show()
     }
+
+    private fun fetchFiles() {
+        if (!checkPermission()) {
+            showPermissionDialog()
+        } else {
+            viewModel.fetchFiles()
+        }
+    }
+
+    private fun showPermissionDialog() {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.R) {
+            val uri = Uri.parse("package:${BuildConfig.APPLICATION_ID}")
+            startActivity(
+                Intent(
+                    Settings.ACTION_MANAGE_APP_ALL_FILES_ACCESS_PERMISSION, uri
+                )
+            )
+        } else requestPermissions(
```

Comment:
Can surround the `requestPermissions` in curly braces as per Java and Kotlin coding standards https://developer.android.com/kotlin/style-guide#braces 
Can take a moment and read the whole article. It's important to get familiar with Kotlin coding standards.

```
else {
    requestPermissions(...)
}
```